src/Persistence/Templates/ControlTemplateStore.cs

```diff
@@ -0,0 +1,33 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System.Diagnostics.CodeAnalysis;
+
+namespace Microsoft.PowerPlatform.PowerApps.Persistence.Templates;
+
+/// <summary>
+/// Control template store.
+/// </summary>
+public class ControlTemplateStore : IControlTemplateStore
+{
+    private readonly SortedList<string, ControlTemplate> _controlTemplatesByName = new();
+    private readonly SortedList<string, ControlTemplate> _controlTemplatesByUri = new();
+
+    public void Add(ControlTemplate controlTemplate)
+    {
+        _ = controlTemplate ?? throw new ArgumentNullException(nameof(controlTemplate));
+
+        _controlTemplatesByName.Add(controlTemplate.Name, controlTemplate);
+        _controlTemplatesByUri.Add(controlTemplate.Uri, controlTemplate);
+    }
+
+    public bool TryGetControlTemplateByName(string name, [NotNullWhen(true)] out ControlTemplate? controlTemplate)
```

Comment:
why two methods to retrieve the template?

Reply:
Made it cleaner and added summary: https://github.com/microsoft/PowerApps-Tooling/pull/537